File: 天梯赛/java/L1-035. 情人节.java
import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int cnt=0;
		String a="",b="";
		while(true) {
			String str=in.nextLine();
			if(str.equals("."))break;
			++cnt;
			if(cnt==2)a=str;
			else if(cnt==14)b=str;
		}
		if(cnt<2)System.out.println("Momo... No one is for you ...");
		else if(cnt<14)System.out.println(a+" is the only one for you...");
		else System.out.println(a+" and "+b+" are inviting you to dinner...");
	}
}

File: 天梯赛/L1-035. 情人节.cpp
#include <iostream> 
#include <cstdio>
#include <set>
#include <cstring>
#include <string>
#include <algorithm>
#include <map>
#include <vector>
#include <queue>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    string tmp,a="",b="";
    int cnt=0;
    while(cin>>tmp&&tmp!="."){
    	++cnt;
    	if(cnt==2)a=tmp;
    	else if(cnt==14)b=tmp;
	}
	if(a.length()&&b.length())cout<<a<<" and "<<b<<" are inviting you to dinner...\n";
	else if(a.length())cout<<a<<" is the only one for you...\n";
	else cout<<"Momo... No one is for you ...\n";
    return 0;
}
